test/lib/bamboo/mailer_test.exs

```diff
@@ -261,6 +193,67 @@ defmodule Bamboo.MailerTest do
     end
   end
 
+  describe "attachments" do
+    defmodule AdapterWithoutAttachmentSupport do
+      def deliver(_email, _config), do: :noop
+
+      def handle_config(config), do: config
+
+      def supports_attachments?, do: false
+    end
+
+    @tag adapter: AdapterWithoutAttachmentSupport
+    test "raise if adapter does not support attachments and attachments are sent" do
+      path = Path.join(__DIR__, "../../support/attachment.docx")
+      email = new_email(to: "foo@bar.com") |> Email.put_attachment(path)
+
+      assert_raise RuntimeError, ~r/does not support attachments/, fn ->
+        Mailer.deliver_now(email)
+      end
+
+      assert_raise RuntimeError, ~r/does not support attachments/, fn ->
+        Mailer.deliver_later(email)
+      end
+    end
+
+    @tag adapter: AdapterWithoutAttachmentSupport
+    test "does not raise if no attachments are on the email" do
+      email = new_email(to: "foo@bar.com")
+      Mailer.deliver_now(email)
+    end
+
+    test "does not raise if adapter supports attachments" do
```

Comment:
Do you think it might be worth it to tag this one explicitly as using the default adapter to distinguish it? Or alternatively move this test up into the section with the tests that use the default adapter?

Reply:
Good suggestion. I think I'll explicitly declare the adapter but leave it in this section as we have a grouping around attachment behavior.